File: RepositoryPattern/SampleApplicationForGM/app/src/main/java/com/example/sampleapplicationforgm/application/BasicAppApplication.java
package com.example.sampleapplicationforgm.application;

import android.app.Activity;
import android.app.Application;
import android.app.Fragment;

import com.example.basicapp.dependencyinjection.DaggerApplicationComponent;

import javax.inject.Inject;

import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.HasActivityInjector;
import dagger.android.HasFragmentInjector;

public class BasicAppApplication extends Application implements HasActivityInjector, HasFragmentInjector {

    @Inject
    DispatchingAndroidInjector<Activity> dispatchingAndroidActivityInjector;
    @Inject
    DispatchingAndroidInjector<Fragment> dispatchingAndroidFragmentInjector;

    @Override
    public void onCreate() {
        super.onCreate();
        buildDaggerComponent();
    }

    private void buildDaggerComponent() {
        DaggerApplicationComponent
                .create()
                .inject(this);
    }

    @Override
    public AndroidInjector<Activity> activityInjector() {
        return dispatchingAndroidActivityInjector;
    }

    @Override
    public AndroidInjector<Fragment> fragmentInjector() {
        return dispatchingAndroidFragmentInjector;
    }
}

File: RepositoryPattern/SampleApplicationForGM/app/src/main/java/com/example/sampleapplicationforgm/application/BasicAppApplication.kt
package com.example.sampleapplicationforgm.application

import android.app.Activity
import android.app.Application
import android.app.Fragment
import com.example.sampleapplicationforgm.dependencyinjection.DaggerApplicationComponent
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasActivityInjector
import dagger.android.HasFragmentInjector
import javax.inject.Inject

class BasicAppApplication : Application(), HasActivityInjector, HasFragmentInjector {
    @JvmField
    @Inject
    var dispatchingAndroidActivityInjector: DispatchingAndroidInjector<Activity>? = null

    @JvmField
    @Inject
    var dispatchingAndroidFragmentInjector: DispatchingAndroidInjector<Fragment>? = null
    override fun onCreate() {
        super.onCreate()
        buildDaggerComponent()
    }

    private fun buildDaggerComponent() {
        DaggerApplicationComponent
                .create()
                .inject(this)
    }

    override fun activityInjector(): AndroidInjector<Activity> {
        return dispatchingAndroidActivityInjector!!
    }

    override fun fragmentInjector(): AndroidInjector<Fragment> {
        return dispatchingAndroidFragmentInjector!!
    }
}